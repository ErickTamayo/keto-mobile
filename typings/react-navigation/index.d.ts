/* eslint-disable */
import ReactNavigation from 'react-navigation'

declare module 'react-navigation' {
  export interface StackNavigatorConfig {
    cardShadowEnabled: boolean
  }
}
