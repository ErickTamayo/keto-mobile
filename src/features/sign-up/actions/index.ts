import { SELECT_GENDER } from '../constants'

export interface SelectGender {
  type: SELECT_GENDER
  gender: 'male' | 'female' | null
}

export const selectGender = (gender: SelectGender['gender']): SelectGender => ({
  type: SELECT_GENDER,
  gender,
})
