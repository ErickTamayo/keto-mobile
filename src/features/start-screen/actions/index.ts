import { ADD_COUNTER, USER_LOGOUT } from '../constants'

export interface AddCounter {
  type: ADD_COUNTER
}

export interface UserLogout {
  type: USER_LOGOUT
}

export const addCounter = (): AddCounter => ({
  type: ADD_COUNTER,
})

export const userLogout = (): UserLogout => ({
  type: USER_LOGOUT,
})
