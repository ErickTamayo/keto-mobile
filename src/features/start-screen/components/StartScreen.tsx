import * as React from 'react'
import { View, Text } from 'react-native'
import { NavigationEvents, NavigationInjectedProps } from 'react-navigation'

import { StateProps, DispatchProps } from '../containers/StartScreen'
import Button from '../../../components/button/Button'
import st from '../../../styles'
import NavigationOptions from '../../../constants/NavigationOptions'

export interface Props extends StateProps, DispatchProps, NavigationInjectedProps {}
export default class StartScreen extends React.Component<Props, object> {
  public static navigationOptions = NavigationOptions

  private didFocus = () => {
    const { userLogout } = this.props
    userLogout()
  }

  private navigateToSignUp = () => {
    const { navigation } = this.props
    navigation.navigate('GenderSelection')
  }

  public render(): JSX.Element {
    return (
      <View style={[st.flex.f1, st.flex.row, st.items.end, st.justify.center, st.bg.greyLightest]}>
        <NavigationEvents onDidFocus={this.didFocus} />
        <View style={[st.flex.f1, st.items.center, st.justify.center, st.m.b10]}>
          <Button title="Create Account" variant="primary-lg" onPress={this.navigateToSignUp} />
          <Button title="Log In" variant="secondary-lg" onPress={() => {}} />
        </View>
      </View>
    )
  }
}
