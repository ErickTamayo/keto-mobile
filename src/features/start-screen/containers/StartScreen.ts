import { connect } from 'react-redux'
import { Dispatch } from 'redux'
import { State } from '../records/StartScreenState'
import StartScreen from '../components/StartScreen'
import { addCounter, AddCounter, userLogout, UserLogout } from '../actions'

export interface StateProps {
  counter: number
}

export interface DispatchProps {
  addCounter: () => AddCounter
  userLogout: () => UserLogout
}

export const mapStateToProps = ({ startScreen }: { startScreen: State }): StateProps => ({
  counter: startScreen.counter,
})

export const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
  addCounter: (): AddCounter => dispatch(addCounter()),
  userLogout: (): UserLogout => dispatch(userLogout()),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(StartScreen)
