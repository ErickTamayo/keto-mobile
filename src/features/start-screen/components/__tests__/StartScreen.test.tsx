import 'react-native'
import * as React from 'react'
import { shallow } from 'enzyme'
import { NavigationScreenProp, NavigationRoute, NavigationParams } from 'react-navigation'
import StartScreen, { Props } from '../StartScreen'
import { AddCounter } from '../../actions'

describe('<StartScreen />', () => {
  it('renders correctly', () => {
    const props: Props = {
      counter: 0,
      addCounter: () => ({} as AddCounter),
      navigation: {} as NavigationScreenProp<NavigationRoute<NavigationParams>, NavigationParams>,
    }
    const wrapper = shallow(<StartScreen {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
