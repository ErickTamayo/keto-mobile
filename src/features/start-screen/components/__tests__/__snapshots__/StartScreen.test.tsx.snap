// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartScreen /> renders correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "flexDirection": "row",
      },
      Object {
        "alignItems": "flex-end",
      },
      Object {
        "backgroundColor": "#f7f7f7",
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "marginBottom": 65,
        },
      ]
    }
  >
    <Component>
      Hello 
      World
      !
    </Component>
    <Component>
      Clicks: 
      0
    </Component>
    <_default
      onPress={[Function]}
      title="Create Account"
      variant="primary-lg"
    />
    <_default
      onPress={[Function]}
      title="Log In"
      variant="secondary-lg"
    />
  </Component>
</Component>
`;
